mixins/visible-on-ancestor/visible-on-ancestor-mixin.js

```diff
@@ -34,6 +45,10 @@ export const VisibleOnAncestorMixin = superclass => class extends superclass {
 
 	static get properties() {
 		return {
+			/**
+			 * @ignore
+			 */
+			animationType: { type: String, reflect: true, attribute: 'animation-type' },
```

Comment:
For the usage of `visible-on-ancestor-mixin` in `button-add`, we only want an opacity animation. The animation is also slightly different than the existing one (300ms ease-in vs 200ms ease-out). This does start to feel a bit custom so let me know if there are alternative ideas for this.

I added the property `animation-type` with the idea there can be different options, since the current default animation is quite specific.

Reply:
Yeah, I have a hard time rationalizing the need for it to be different, especially the timing function. Like some other use-case could come along and want something else. But I'm sure Jeff has weighed the options and wouldn't ask for it if it wasn't needed.

Another option would be to expose CSS variables that consumers could override. But I think I like keeping it under wraps as much as possible, and not exposing specific CSS details to consumers that would make it difficult for us to change the animations later.